fuse/dl/models/backbones/backbone_resnet.py

```diff
@@ -33,15 +33,18 @@ def __init__(
         self,
         *,
         pretrained: bool = False,
-        weights: Optional[WeightsEnum] = None,
+        weights: Optional[Union[WeightsEnum, dict]] = None,
         in_channels: int = 3,
-        name: str = "resnet18"
+        name: str = "resnet18",
+        pool: bool = False,
     ) -> None:
         """
         Create 2D Resnet
         :param pretrained: reload imagenet weights
         :param in_channels: Number of input channels
         :param name: model name. Currently support 'resnet18' and 'resnet50'
+        :param pool: whether to use global average pooling to reduce the spacial dimensions after the convolutional layers
```

Comment:
So it's always AvgPool, do you think that pool should be a string and support also max pooling?